internal/resources/azure/frontdoor.go

```diff
@@ -0,0 +1,324 @@
+package azure
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/infracost/infracost/internal/usage"
+	"github.com/shopspring/decimal"
+)
+
+// Frontdoor struct represents Azure's Front Door network service.
+//
+// More resource information here: https://docs.microsoft.com/en-us/azure/frontdoor/front-door-overview
+// Pricing information here: https://azure.microsoft.com/en-us/pricing/details/frontdoor/ (Azure Front Door tab)
+type Frontdoor struct {
+	Address string
+	Region  string
+
+	FrontendHostsQuantity int
```

Comment:
We don't add `Quantity` suffixes elsewhere. I'm open to changing this, just thought I'd flag up this inconsistency.

Reply:
For context: I wasn't sure if it's better to pass the whole array and calculate the length here. But decided that this struct should be aware of information related to pricing only which is quantity in this case. So went with this name to distinguish array and int values. I've also seen some examples using `numberOfX`. But with Go's types it should be simple to recognize it I guess. Will change to `FrontendHosts`/`RoutingRules` style. 👍